indico/modules/rb_new/operations/rooms.py

```diff
@@ -191,21 +192,21 @@ def search_for_rooms(filters, availability=None):
     include_blockings = not rb_is_admin(session.user)
     availability_filters = [Room.filter_available(start_dt, end_dt, repeatability, include_blockings=include_blockings,
                                                   include_pre_bookings=True)]
-    if not rb_is_admin(session.user):
+    if not (admin_override_enabled and rb_is_admin(session.user)):
         # TODO: apply this filter for each room based on the ACL
         # only query whether the restriction is violated or not and return this
         # information to the client (or filter here based on can_override for each room)
         # doing it on the client is probably better since it has all this information cached
         selected_period_days = (filters['end_dt'] - filters['start_dt']).days
         booking_limit_days = db.func.coalesce(Room.booking_limit_days, rb_settings.get('booking_limit'))
 
-        own_rooms = [r.id for r in Room.get_owned_by(session.user)]
-        criterion = db.or_(Room.id.in_(own_rooms) if own_rooms else False,
-                           db.and_(Room.filter_bookable_hours(start_dt.time(), end_dt.time()),
-                                   Room.filter_nonbookable_periods(start_dt, end_dt),
-                                   db.or_(booking_limit_days.is_(None),
-                                          selected_period_days <= booking_limit_days)))
-        availability_filters.append(criterion)
+        criterion = db.and_(Room.filter_bookable_hours(start_dt.time(), end_dt.time()),
+                            Room.filter_nonbookable_periods(start_dt, end_dt),
+                            db.or_(booking_limit_days.is_(None),
+                            selected_period_days <= booking_limit_days))
+        cant_book = [room.id for room in query.filter(db.and_(*availability_filters)).filter(~criterion).all() if
```

Comment:
That looks a bit unreadable, what about sth like this?

```python

cant_book = [room.id
             for room in query.filter(db.and_(*availability_filters)).filter(~criterion).all()
             if not room.can_override(session.user, allow_admin=False)]
```

Reply:
:+1: wrapping after `if` doesn't look amazing

Also, the querying part can be simplified to `query.filter(db.and_(*availability_filters), ~criterion)`